File: my-submissions/m739.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        output = [0] * len(temperatures)
        tempStack = []                      # schema: (temperature, index)

        for i in range(len(temperatures) - 1, -1, -1) :
            while tempStack :
                if tempStack[-1][0] > temperatures[i] :
                    output[i] = tempStack[-1][1] - i
                    break
                else :
                    tempStack.pop()
            
            tempStack.append((temperatures[i], i))

        return output

File: my-submissions/m739.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<int[]> previousMaxes = new Stack<>();  // Schema (temp, indx)

        int[] output = new int[temperatures.length];

        for (int i = temperatures.length - 1; i >= 0; i--) {
            while (!previousMaxes.isEmpty()) {
                if (previousMaxes.peek()[0] > temperatures[i]) {
                    output[i] = previousMaxes.peek()[1] - i;
                    break;
                }
                previousMaxes.pop();
            }
            previousMaxes.push(new int[] {temperatures[i], i});
        }

        return output;
    }
}